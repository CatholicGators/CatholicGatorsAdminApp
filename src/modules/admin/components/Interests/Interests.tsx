import React, { Component } from 'react'
import { connect } from 'react-redux';
import {
    Paper,
    Typography,
    Toolbar,
    Theme,
    createStyles,
    withStyles,
    CircularProgress,
    Button
} from '@material-ui/core'
import EditableInterestOption from './components/EditableInterestOption/EditableInterestOption';
import {
    getInterests,
    updateInterests,
    addOption
} from '../../../../redux/actions/contactForm/interestActions';
import AddOptionButton from './components/AddOptionButton/AddOptionButton';

const styles = (theme: Theme) => createStyles({
    formWrapper: {
        width: 600
    },
    form: {
        margin: `0 ${theme.spacing.unit * 3}px ${theme.spacing.unit * 3}px ${theme.spacing.unit * 3}px`
    },
    formLoadingContainer: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        minHeight: '300px',
    },
    addSectionBtn: {
        marginLeft: 'auto'
    }
})

export type Props = {
    classes: any,
    interests: Section[]
    getInterests: () => void,
    updateInterests: (interests : Section[]) => void,
    addOption: (sectionId: string, option: Option) => void
}

type State = {
    editingOptionId: number | null
    addingSectionId: string
}

export type Section = {
    id: string
    text: string
    options: Option[]
}

export type Option = {
    id?: number
    sectionId: string
    text: string
}

export class Interests extends Component<Props, State> {
    state = {
        editingOptionId: null,
        addingSectionId: null
    }

    componentDidMount() {
        this.props.getInterests()
    }

    beginEditingOption(optionId: number) {
        this.setState({ editingOptionId: optionId, addingSectionId: null })
    }

    cancelEditingOption() {
        this.setState({ editingOptionId: null })
    }

    deleteOption(sectionId: string, optionId: number) {
        this.setState({ editingOptionId: null })
    }

    saveOption(sectionId: string, optionId: number, newText: string) {
        this.props.updateInterests([
            ...this.props.interests.map(section => section.id !== sectionId ? section : {
                ...section,
                options: [
                    ...section.options.map(option => option.id !== optionId ? option : {
                        ...option,
                        text: newText
                    })
                ]
            })
        ])
        this.setState({ editingOptionId: null })
    }

    beginAddingOption(sectionId: string) {
        this.setState({ addingSectionId: sectionId, editingOptionId: null })
    }   

    cancelAddingOption() {
        this.setState({ addingSectionId: null })
    }

    addOption(sectionId: string, text: string) {
        this.props.addOption(sectionId, { sectionId, text })
        this.setState({ addingSectionId: null })
    }

    render() {
        const { classes, interests } = this.props
        const { addingSectionId, editingOptionId } = this.state
        return (
            <Paper className={classes.formWrapper}>
                <Toolbar>
                    <Typography variant="h6">Edit Interests step of form</Typography>
                    <Button variant="outlined" color="primary" className={classes.addSectionBtn}>
                        Add Section
                    </Button>
                </Toolbar>
                {interests ? (
                    <div className={classes.form}>
                        {interests.map(section => 
                            <div key={section.id}>
                                <Typography>{section.text}</Typography>
                                {section.options.map(option =>
                                    <EditableInterestOption
                                        key={option.id}
                                        isEditing={option.id === editingOptionId}
                                        option={option}
                                        beginEditing={optionId => this.beginEditingOption(optionId)}
                                        deleteOption={optionId => this.deleteOption(section.id, optionId)}
                                        saveOption={(optionId, newText) => this.saveOption(section.id, optionId, newText)}
                                        cancelEditing={() => this.cancelEditingOption()}
                                    />
                                )}
                                <AddOptionButton
                                    isAdding={addingSectionId === section.id}
                                    beginAddingOption={() => this.beginAddingOption(section.id)}
                                    cancelAddingOption={() => this.cancelAddingOption()}
                                    addOption={text => this.addOption(section.id, text)}
                                />
                            </div>
                        )}
                    </div>
                ) : (
                    <div className={classes.formLoadingContainer}>
                        <CircularProgress size="60px" />
                    </div>
                )}
            </Paper>
        )
    }
}

const mapStateToProps = state => ({
    interests: state.contactForm.interests
})

const mapDispatchToProps = dispatch => ({
    getInterests: () => dispatch(getInterests()),
    updateInterests: interests => dispatch(updateInterests(interests)),
    addOption: (sectionId: string, option: Option) => dispatch(addOption(sectionId, option))
})

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Interests))
