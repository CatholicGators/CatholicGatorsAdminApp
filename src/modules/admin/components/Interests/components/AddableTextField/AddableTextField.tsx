import React, { Component } from 'react'
import {
    Theme,
    createStyles,
    withStyles,
    IconButton,
    Input
} from '@material-ui/core'
import { Save, Close, Add } from '@material-ui/icons'

export const styles = (theme: Theme) =>
    createStyles({
        checkbox: {
            margin: 0,
            pointerEvents: 'none'
        },
        input: {
            margin: 0
        },
        row: {
            display: 'flex',
            alignItems: 'center',
            '&:hover': {
                '& section': {
                    display: 'block'
                }
            },
            width: '100%'
        },
        actions: {
            display: 'none',
            marginLeft: `${theme.spacing.unit}px`
        },
        editActions: {
            display: 'flex',
            alignItems: 'center'
        }
    })

export type Props = {
    isAdding: boolean
    classes: any
    beginAdding: () => void
    onAdd: (text: string) => void
    cancelAdding: () => void
}

type State = {
    text: string
}

export class AddableTextField extends Component<Props, State> {
    state = {
        text: ''
    }

    onChange(text: string) {
        this.setState({ text })
    }

    add() {
        const { text } = this.state
        if (text) {
            this.props.onAdd(text)
            this.setState({ text: '' })
        }
    }

    render() {
        const { isAdding, classes, beginAdding, cancelAdding } = this.props

        const { text } = this.state

        return isAdding ? (
            <div className={classes.row}>
                <Input
                    fullWidth
                    className={classes.input}
                    value={text}
                    onChange={ev => this.onChange(ev.target.value)}
                />
                <div className={classes.editActions}>
                    <IconButton
                        id="save-btn"
                        disabled={!text}
                        onClick={() => this.add()}
                    >
                        <Save />
                    </IconButton>
                    <IconButton id="close-btn" onClick={cancelAdding}>
                        <Close />
                    </IconButton>
                </div>
            </div>
        ) : (
            <IconButton onClick={beginAdding}>
                <Add />
            </IconButton>
        )
    }
}

export default withStyles(styles)(AddableTextField)
