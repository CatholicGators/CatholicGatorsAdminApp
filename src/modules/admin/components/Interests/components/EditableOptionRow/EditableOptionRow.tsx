import React, { Component } from 'react'
import {
    Checkbox,
    withStyles,
    createStyles,
    Theme
} from '@material-ui/core';
import EditableTextField from '../EditableTextField/EditableTextField';
import { Option } from '../../../../../../services/interestsService';

const styles = (theme: Theme) => createStyles({
    checkbox: {
        margin: 0,
        pointerEvents: 'none'
    },
    optionRow: {
        display: 'flex',
        alignItems: 'center',
        '&:hover': {
            'section': {
                display: 'block'
            }
        }
    }
})

export type Props = {
    classes: any,
    option: Option,
    editingOptionId: number,
    beginEditingOption: (optionId: any) => void,
    cancelEditingOption: () => void,
    deleteOption: (optionId: any) => void,
    saveOption: (optionId: any, newText: string) => void
}

export type State = {
    isHovered: boolean
}

export class EditableOptionRow extends Component<Props, State> {
    state = {
        isHovered: false
    }

    render() {
        const {
            option,
            editingOptionId,
            classes
        } = this.props
        const { isHovered } = this.state

        return (
            <div
                className={classes.optionRow}
                onMouseEnter={() => this.setState({ isHovered: true })}
                onMouseLeave={() => this.setState({ isHovered: false })}
            >
                <Checkbox
                    className={classes.checkbox}
                    color='primary'
                />
                <EditableTextField
                    isHovered={isHovered}
                    isEditing={option.id === editingOptionId}
                    id={option.id}
                    text={option.text}
                    beginEditing={optionId => this.props.beginEditingOption(optionId)}
                    deleteText={optionId => this.props.deleteOption(optionId)}
                    save={(optionId, newText) => this.props.saveOption(optionId, newText)}
                    cancelEditing={() => this.props.cancelEditingOption()}
                />
            </div>
        )
    }
}

export default withStyles(styles)(EditableOptionRow)