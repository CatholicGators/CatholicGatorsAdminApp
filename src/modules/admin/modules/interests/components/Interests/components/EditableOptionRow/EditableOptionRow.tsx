import React, { Component } from "react"
import { Checkbox, withStyles, createStyles, Theme } from "@material-ui/core"
import EditableTextField from "../EditableTextField/EditableTextField"
import { Option } from "../../../../services/interestsService"

export const styles = (theme: Theme) =>
    createStyles({
        checkbox: {
            margin: 0,
            pointerEvents: "none",
        },
        optionRow: {
            display: "flex",
            alignItems: "center",
            "&:hover": {
                section: {
                    display: "block",
                },
            },
            height: "48px",
        },
    })

export type Props = {
    classes: any
    option: Option
    editingOptionId: string
    beginEditingOption: (optionId: string) => void
    cancelEditingOption: () => void
    deleteOption: (optionId: string) => void
    saveOption: (optionId: string, newText: string) => void
}

export type State = {
    isHovered: boolean
}

export class EditableOptionRow extends Component<Props, State> {
    state = {
        isHovered: false,
    };

    render() {
        const { option, editingOptionId, classes } = this.props
        const { isHovered } = this.state

        return (
            <div
                className={classes.optionRow}
                onMouseEnter={() => this.setState({ isHovered: true })}
                onMouseLeave={() => this.setState({ isHovered: false })}
            >
                <Checkbox className={classes.checkbox} color="primary" />
                <EditableTextField
                    isHovered={isHovered}
                    isEditing={option.id === editingOptionId}
                    id={option.id}
                    text={option.text}
                    beginEditing={this.props.beginEditingOption}
                    deleteText={this.props.deleteOption}
                    save={this.props.saveOption}
                    cancelEditing={this.props.cancelEditingOption}
                />
            </div>
        )
    }
}

export default withStyles(styles)(EditableOptionRow)
