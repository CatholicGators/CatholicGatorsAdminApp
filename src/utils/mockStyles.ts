import { createMuiTheme } from '@material-ui/core'

export default (styles) => Object.keys(styles(createMuiTheme())).reduce((acc, curr) => ({...acc, [curr]: "test"}), {})