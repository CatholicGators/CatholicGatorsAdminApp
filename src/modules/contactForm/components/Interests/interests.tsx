import React from "react";
import { styles } from '../../utils/ContactFormStyles'

import {
  withStyles,
  Checkbox,
  FormControlLabel,
  FormLabel,
  FormGroup,
  Grid,
  Button
} from '@material-ui/core';

class Interests extends React.Component {
  constructor(public props: any) {
    super(props)
  }

  render() {
    const { classes } = this.props;
    const handleChange = this.props.handleChange;

    return (
      <React.Fragment>
        <form onSubmit={this.props.handleNext}>
            <FormLabel>I am interested in...</FormLabel>
            <FormGroup>
            <FormControlLabel
                control={
                <Checkbox
                    checked={this.props.data.eventsGreekStudents}
                    onChange={handleChange('eventsGreekStudents')}
                    value="eventsGreekStudents"
                ></Checkbox>
                }
                label="events for Greek students"
            />
            <FormControlLabel
                control={
                <Checkbox
                    checked={this.props.data.eventsLatinoStudents}
                    onChange={handleChange('eventsLatinoStudents')}
                    value="eventsLatinoStudents"
                ></Checkbox>
                }
                label="events for Latino/a students"
            />
            <FormControlLabel
                control={
                <Checkbox
                    checked={this.props.data.eventsGraduateStudents}
                    onChange={handleChange('eventsGraduateStudents')}
                    value="eventsGraduateStudents"
                ></Checkbox>
                }
                label="events for graduate students"
            />
            </FormGroup>
            <br/>
            <FormLabel className={classes.margins}>I would like to...</FormLabel>
            <FormGroup>
            <FormControlLabel
                control={
                <Checkbox
                    checked={this.props.data.receiveMonthlyNewsletter}
                    onChange={handleChange('receiveMonthlyNewsletter')}
                    value="receiveMonthlyNewsletter"
                ></Checkbox>
                }
                label="receive the monthly e-newsletter"
            />
            <FormControlLabel
                control={
                <Checkbox
                    checked={this.props.data.registerAsParishioner}
                    onChange={handleChange('registerAsParishioner')}
                    value="registerAsParishioner"
                ></Checkbox>
                }
                label="register as a parishioner"
            />
            </FormGroup>
            <br/>
            <FormLabel className={classes.margins}>
            I am interested in getting involved in...
            </FormLabel>
            <FormGroup>
            <Grid container direction="row">
                <Grid item className={classes.margins}>
                <Grid container direction="column">
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.englishBibleStudy}
                        onChange={handleChange('englishBibleStudy')}
                        value="englishBibleStudy"
                        ></Checkbox>
                    }
                    label="English Bible Studys"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.spanishBibleStudy}
                        onChange={handleChange('spanishBibleStudy')}
                        value="spanishBibleStudy"
                        ></Checkbox>
                    }
                    label="Spanish/Bilingual Bible Study"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.freeFood}
                        onChange={handleChange('freeFood')}
                        value="freeFood"
                        ></Checkbox>
                    }
                    label="Free Food"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.guestSpeakers}
                        onChange={handleChange('guestSpeakers')}
                        value="guestSpeakers"
                        ></Checkbox>
                    }
                    label="Guest Speakers"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.musicMinistry}
                        onChange={handleChange('musicMinistry')}
                        value="musicMinistry"
                        ></Checkbox>
                    }
                    label="Music Ministry"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.socials}
                        onChange={handleChange('socials')}
                        value="socials"
                        ></Checkbox>
                    }
                    label="Socials"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.retreats}
                        onChange={handleChange('retreats')}
                        value="retreats"
                        ></Checkbox>
                    }
                    label="Retreats"
                    />
                </Grid>
                </Grid>
                <Grid item className={classes.margins}>
                <Grid container direction="column">
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.intramuralSports}
                        onChange={handleChange('intramuralSports')}
                        value="intramuralSports"
                        ></Checkbox>
                    }
                    label="Intramural Sports"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.proLifeClub}
                        onChange={handleChange('proLifeClub')}
                        value="proLifeClub"
                        ></Checkbox>
                    }
                    label="Pro-Life Club"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.RHC}
                        onChange={handleChange('RHC')}
                        value="RHC"
                        ></Checkbox>
                    }
                    label="Communications (Photo, Print, Graphics, PR, Advertising, and Video)"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.servingAtMass}
                        onChange={handleChange('servingAtMass')}
                        value="servingAtMass"
                        ></Checkbox>
                    }
                    label="Serving at Mass"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.teachReligiousEd}
                        onChange={handleChange('teachReligiousEd')}
                        value="teachReligiousEd"
                        ></Checkbox>
                    }
                    label="Teach Religious Education"
                    />
                    <FormControlLabel
                    control={
                        <Checkbox
                        checked={this.props.data.serviceProjects}
                        onChange={handleChange('serviceProjects')}
                        value="serviceProjects"
                        ></Checkbox>
                    }
                    label="Service Projects and Trips"
                    />
                </Grid>
                </Grid>
            </Grid>
            </FormGroup>
            <div className={classes.buttons}>
                <Button
                    onClick={this.props.handleBack}
                    className={classes.button}
                > Back </Button>
                <Button
                    variant="contained"
                    color="primary"
                    type="submit"
                    className={classes.button}
                > Submit form </Button>
              </div>
        </form>
      </React.Fragment>
    )
  }
}

export default withStyles(styles)(Interests);
