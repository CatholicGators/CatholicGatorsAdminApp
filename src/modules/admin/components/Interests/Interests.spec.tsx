import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { Interests, Props, defaultState, styles } from './Interests'
import { CircularProgress } from '@material-ui/core'
import EditableOptionRow from './components/EditableOptionRow/EditableOptionRow'
import { Checkbox } from '@material-ui/core'
import AddableTextField from './components/AddableTextField/AddableTextField'
import mockStyles from '../../../../utils/mockStyles'

describe('Interests', () => {
    let props: Props, wrapper

    beforeEach(() => {
        props = {
            classes: mockStyles(styles),
            interests: [
                {
                    id: 'test',
                    position: 0,
                    text: 'testing',
                    options: [
                        {
                            id: '1',
                            text: 'totally testing'
                        }
                    ]
                }
            ],
            getInterests: jest.fn(),
            updateOptionText: jest.fn(),
            addOption: jest.fn(),
            addSection: jest.fn()
        }
        wrapper = shallow(<Interests {...props} />)
    })

    it('should match snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('calls getInterests() on componentDidMount', () => {
        expect(props.getInterests).toHaveBeenCalled()
    })

    it('renders a spinner when interests is falsy', () => {
        wrapper.setProps({ interests: undefined })
        expect(wrapper.exists(CircularProgress)).toBe(true)
    })

    it('is not adding or editing an item by default', () => {
        expect(wrapper.state().editingOptionId).toBe(null)
        expect(wrapper.state().addingSectionId).toBe(null)
        expect(wrapper.state().isAddingSection).toBe(false)
    })

    describe('option crud', () => {
        let optionId,
            row,
            newText = 'newText'
        beforeEach(() => {
            optionId = props.interests[0].options[0].id
            row = wrapper
                .find('#section')
                .first()
                .find(EditableOptionRow)
                .first()

            row.props().beginEditingOption(optionId)
        })

        describe('beginEditingOption', () => {
            it('sets the editingOptionId to the id of the option being edited', () => {
                expect(wrapper.state()).toEqual({
                    ...defaultState,
                    editingOptionId: optionId
                })
            })
        })

        describe('cancelEditingOption', () => {
            it('resets the editing state to default', () => {
                row.props().cancelEditingOption()
                expect(wrapper.state()).toEqual(defaultState)
            })
        })

        describe('deleteOption', () => {
            it('resets the editing state to default', () => {
                row.props().deleteOption(optionId)
                expect(wrapper.state()).toEqual(defaultState)
            })
        })

        describe('saveOption', () => {
            it('resets the editing state to default', () => {
                row.props().saveOption(optionId, newText)
                expect(wrapper.state()).toEqual(defaultState)
            })

            it('calls updateInterests with the field updated', () => {
                row.props().saveOption(optionId, newText)
                expect(props.updateOptionText).toHaveBeenCalledWith(
                    optionId,
                    newText
                )
            })
        })
    })

    describe('section crud', () => {
        let addOptioncomponent

        beforeEach(() => {
            addOptioncomponent = wrapper
                .find('#section')
                .first()
                .find('#add-option')
                .first()
        })

        it('doesnt show the check box when addingSectionId === section.id', () => {
            expect(wrapper.exists(Checkbox)).toBe(false)
        })

        describe('add section button', () => {
            it('sets the state to adding a section on click and shows the add section input input', () => {
                wrapper.find('#add-section-btn').simulate('click')

                expect(wrapper.state()).toEqual({
                    ...defaultState,
                    isAddingSection: true
                })
                expect(wrapper.exists('#add-section-input')).toBe(true)
            })
        })

        describe('adding an option to a section', () => {
            let textField, sectionId

            beforeEach(() => {
                sectionId = props.interests[0].id
                textField = addOptioncomponent.find(AddableTextField)
                textField.props().beginAdding()
            })

            it('sets addingSectionId to the id of the section being added to', () => {
                expect(wrapper.state()).toEqual({
                    ...defaultState,
                    addingSectionId: sectionId
                })
            })

            it('shows a checkbox', () => {
                expect(wrapper.exists(Checkbox)).toBe(true)
            })

            it('resets the state when adding cancelled', () => {
                textField.props().cancelAdding()

                expect(wrapper.state()).toEqual(defaultState)
            })

            it('calls addOption when option saved and resets state', () => {
                const newText = 'newText'

                textField.props().onAdd(newText)

                expect(props.addOption).toHaveBeenCalledWith(sectionId, {
                    text: newText
                })
                expect(wrapper.state()).toEqual(defaultState)
            })
        })
    })
})
