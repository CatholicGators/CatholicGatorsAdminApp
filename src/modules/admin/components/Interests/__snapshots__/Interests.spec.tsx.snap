// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interests should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Interests
    classes={Object {}}
    getInterests={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    interests={
      Array [
        Object {
          "id": "test",
          "options": Array [
            Object {
              "id": 1,
              "text": "totally testing",
            },
          ],
          "text": "testing",
        },
      ]
    }
    updateInterests={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(Toolbar)>
          <WithStyles(Typography)
            variant="h6"
          >
            Edit Interests step of form
          </WithStyles(Typography)>
        </WithStyles(Toolbar)>,
        <div>
          <div>
            <WithStyles(Typography)>
              testing
            </WithStyles(Typography)>
            <WithStyles(EditableInterestOption)
              beginEditing={[Function]}
              cancelEditing={[Function]}
              deleteOption={[Function]}
              isEditing={false}
              option={
                Object {
                  "id": 1,
                  "text": "totally testing",
                }
              }
              saveOption={[Function]}
            />
          </div>
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            variant="h6"
          >
            Edit Interests step of form
          </WithStyles(Typography)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Edit Interests step of form",
            "variant": "h6",
          },
          "ref": null,
          "rendered": "Edit Interests step of form",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <WithStyles(Typography)>
                testing
              </WithStyles(Typography)>
              <WithStyles(EditableInterestOption)
                beginEditing={[Function]}
                cancelEditing={[Function]}
                deleteOption={[Function]}
                isEditing={false}
                option={
                  Object {
                    "id": 1,
                    "text": "totally testing",
                  }
                }
                saveOption={[Function]}
              />
            </div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "test",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)>
                  testing
                </WithStyles(Typography)>,
                Array [
                  <WithStyles(EditableInterestOption)
                    beginEditing={[Function]}
                    cancelEditing={[Function]}
                    deleteOption={[Function]}
                    isEditing={false}
                    option={
                      Object {
                        "id": 1,
                        "text": "totally testing",
                      }
                    }
                    saveOption={[Function]}
                  />,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "testing",
                },
                "ref": null,
                "rendered": "testing",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "class",
                "props": Object {
                  "beginEditing": [Function],
                  "cancelEditing": [Function],
                  "deleteOption": [Function],
                  "isEditing": false,
                  "option": Object {
                    "id": 1,
                    "text": "totally testing",
                  },
                  "saveOption": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Toolbar)>
            <WithStyles(Typography)
              variant="h6"
            >
              Edit Interests step of form
            </WithStyles(Typography)>
          </WithStyles(Toolbar)>,
          <div>
            <div>
              <WithStyles(Typography)>
                testing
              </WithStyles(Typography)>
              <WithStyles(EditableInterestOption)
                beginEditing={[Function]}
                cancelEditing={[Function]}
                deleteOption={[Function]}
                isEditing={false}
                option={
                  Object {
                    "id": 1,
                    "text": "totally testing",
                  }
                }
                saveOption={[Function]}
              />
            </div>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              variant="h6"
            >
              Edit Interests step of form
            </WithStyles(Typography)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Edit Interests step of form",
              "variant": "h6",
            },
            "ref": null,
            "rendered": "Edit Interests step of form",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <WithStyles(Typography)>
                  testing
                </WithStyles(Typography)>
                <WithStyles(EditableInterestOption)
                  beginEditing={[Function]}
                  cancelEditing={[Function]}
                  deleteOption={[Function]}
                  isEditing={false}
                  option={
                    Object {
                      "id": 1,
                      "text": "totally testing",
                    }
                  }
                  saveOption={[Function]}
                />
              </div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "test",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <WithStyles(Typography)>
                    testing
                  </WithStyles(Typography)>,
                  Array [
                    <WithStyles(EditableInterestOption)
                      beginEditing={[Function]}
                      cancelEditing={[Function]}
                      deleteOption={[Function]}
                      isEditing={false}
                      option={
                        Object {
                          "id": 1,
                          "text": "totally testing",
                        }
                      }
                      saveOption={[Function]}
                    />,
                  ],
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "testing",
                  },
                  "ref": null,
                  "rendered": "testing",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "class",
                  "props": Object {
                    "beginEditing": [Function],
                    "cancelEditing": [Function],
                    "deleteOption": [Function],
                    "isEditing": false,
                    "option": Object {
                      "id": 1,
                      "text": "totally testing",
                    },
                    "saveOption": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
